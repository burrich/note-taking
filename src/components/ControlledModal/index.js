import React from 'react';
import { Modal } from 'semantic-ui-react';

/**
 * Semantic UI controlled modal component.
 */
function ControlledModal(props) {
  return (
    <Modal 
      open={props.open}
      onClose={props.onClose}
      size={props.size}>

      <Modal.Header>{props.header}</Modal.Header>

      <Modal.Content>{props.content}</Modal.Content>

      {props.modalActions}
    </Modal>
  );
}

export default ControlledModal;
