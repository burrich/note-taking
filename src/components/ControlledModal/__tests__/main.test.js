import React from 'react';
import { shallow } from 'enzyme'

import ControlledModal from '../';

test('Controlled modal renders correctly', () => {
  const modal = shallow(
    <ControlledModal
      open={true}
      onClose={jest.fn()}
      size="mini"
      header="controlled modal"
      content="modal content"
      modalActions="modal action" />
  );
  expect(modal).toMatchSnapshot();
});