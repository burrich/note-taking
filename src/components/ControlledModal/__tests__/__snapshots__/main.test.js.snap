// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controlled modal renders correctly 1`] = `
<Modal
  centered={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[Function]}
  open={true}
  size="mini"
>
  <ModalHeader>
    controlled modal
  </ModalHeader>
  <ModalContent>
    modal content
  </ModalContent>
  modal action
</Modal>
`;
